xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-image-old/plugin.js

```diff
@@ -190,6 +190,14 @@
               (newResourceReference.type !== 'url' || newResourceReference.reference.indexOf('://') < 0);
             widget.setData('resourceReference', CKEDITOR.tools.extend(newResourceReference, oldResourceReference));
           }
+        },
+        getBase: function () {
+          var currentInstance = CKEDITOR.currentInstance;
+          var base;
+          if (currentInstance) {
+            base = currentInstance.config.sourceDocument.documentReference;
+          }
+          return base;
```

Comment:
Why not put this directly in ``xwiki-resource`` plugin? I see you duplicate it also in the link plugin, but I'm not sure why.

Reply:
I'm not sure to understand, I thought you said in previous comments that dependencies on CKEditor should be avoided in `xwiki-resource`. Unless you meant that `xwiki-resource` contains  some generic part (that could be extracted in a separate module), and the "glue" part (i.e., `xwiki-resource/plugins.js`). In that case I agree it would be better to move `getBase` here and remove the duplication.

Reply:
> Unless you meant that xwiki-resource contains some generic part (that could be extracted in a separate module), and the "glue" part (i.e., xwiki-resource/plugins.js).

Yes, ``xwiki-resource/plugins.js`` is of course specific to CKEditor, as it defines a CKEditor plugin, so it has to use the CKEditor API. The rest, ``resource*.js`` and ``entityResource*.js``, are generic, defining the "resource picker", that can be used outside of CKEditor. Sorry for the confusion.

Reply:
No worry, now that you say it it's kind of obvious :)